### ISSUE TITLE
Uncaught TypeError: editor.displayBuffer.getScrollTop is not a function

### ISSUE THREAD

**Issue by mybruce** [TypeStrong/atom-typescript#647]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.2.0-dev-5a4f79c
**System**: Mac OS X 10.11
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v7.1.5
### Stack Trace

Uncaught TypeError: editor.displayBuffer.getScrollTop is not a function

```
At /Users/brucex/.atom/packages/atom-typescript/dist/main/atom/tooltipManager.js:49

TypeError: editor.displayBuffer.getScrollTop is not a function
    at showExpressionType (/Users/brucex/.atom/packages/atom-typescript/dist/main/atom/tooltipManager.js:49:45)
    at /Users/brucex/.atom/packages/atom-typescript/dist/main/atom/tooltipManager.js:40:59
```
### Commands

```
  4x -1:02.4.0 tree-view:toggle (atom-text-editor.editor.is-focused)
  3x -0:58.4.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:54.1.0 editor:delete-to-beginning-of-line (atom-text-editor.editor.is-focused)
     -0:45.3.0 tree-view:toggle (atom-pane.pane.active)
     -0:12.3.0 core:move-down (atom-text-editor.editor.is-focused)
     -0:11.8.0 editor:move-to-end-of-screen-line (atom-text-editor.editor.is-focused)
     -0:11.4.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:05.7.0 core:backspace (atom-text-editor.editor.is-focused)
  4x -0:04.1.0 editor:delete-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -0:03.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:02.6.0 editor:delete-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -0:02.2.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:01.8.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "vim-mode"
    ]
  },
  "atom-typescript": {}
}
```
### Installed Packages

``` coffee
# User
.bin, vundefined
atom-typescript, v7.1.5
hyperclick, v0.0.32
linter, v1.8.0

# Dev
No dev packages
```

**Comment by basarat** [comment]
> Atom Version: 1.2.0-dev-5a4f79c

Perhaps a bug in beta software? Would need a repro with a supported version. Thanks :rose:

**Comment by KapuzenSohn** [comment]
Got the same error. 
I am using Atom `1.1.0-beta.1`
Plugin version is: `7.5.0`

The error opens as soon as I open a .ts file. Even more errors show up when I move the mouse.
I tried reinstalling the plugin and also build atom new

**Comment by basarat** [comment]
We now have two reports on beta channel .... getting self ready for lots of complaints after beta becomes main :) **deep breaths**

**Comment by DCStudios** [comment]
Same error.
Atom version 1.1.0-beta.1
atom-typescript version 7.5.1

As mentioned, error pops when open a .ts file,
when mouse move, but also each time I click.
The error says :
```
   editor.displayBuffer.getScrollTop is not a function
```

Because any mouse move pops a new error bubble,
working with the plugin is impossible,

**Comment by TheTedAdams** [comment]
Just installed atom-typescript from apm on a fresh installation of Atom 1.1.0 and am observing this error. Any mouse movement over the file causes atom-typescript to attempt to call this function in tooltipManager.

**Comment by fcoury** [comment]
Submitted a pull request that fixes this. Apparently `getScrollTop()` and `getScrollLeft()` was moved to the `editor`.

**Comment by enriched** [comment]
Dang, beat us to it.

**Comment by fcoury** [comment]
@enriched :grinning:

**Comment by fcoury** [comment]
Quick fix for people coming here from Atom 1.1.0:

```
sed -i '' 's/editor.displayBuffer.getScroll/editor.getScroll/g' ~/.atom/packages/atom-typescript/dist/main/atom/tooltipManager.js
```

Then quit and restart Atom.

**Comment by TheTedAdams** [comment]
Can confirm fcoury's sed fix works.

Found the Commit where they removed the method from display-buffer:
https://github.com/atom/atom/commit/535a9da94657f92498f8bc7f3f27c07c779555d6

**Comment by basarat** [comment]
released https://github.com/TypeStrong/atom-typescript/releases/tag/v7.6.4 :rose:

**Comment by mrkishi** [comment]
@basarat: The fix is not actually included in the release.

**Comment by basarat** [comment]
Indeed. The PR : https://github.com/TypeStrong/atom-typescript/commit/8a6368e7f1930866a2c2c3580a7c2e98ec2dd928 was on a `.js` file! I'll do a new release (and test it still works) when I put out some other fires ....

**Comment by basarat** [comment]
Fixed with https://github.com/TypeStrong/atom-typescript/releases/tag/v7.6.5 :rose: Phew https://github.com/TypeStrong/atom-typescript/compare/v7.6.4...v7.6.5

**Comment by fcoury** [comment]
Sorry for the noob mistake @basarat @mrkishi

**Comment by AngelMunoz** [comment]
hello! It's happening to me in version 7.7.0  with atom 1.10 :(
reinstalled twice :(

**Comment by jbeck111** [comment]
And me. Sorry guys but it's not fixed. Everything up to date here with 7.7.0 and atom 1.10.

**Comment by basarat** [comment]
@AngelMunoz @jbeck111 are you sure its this package and not *minimap* or something else?

**Comment by AngelMunoz** [comment]
Well, I'll try first hour in the morning,  with the minimap uninstalled not able to test right now :(

Sent from Outlook




On Fri, Oct 30, 2015 at 9:38 PM -0700, "Basarat Ali Syed" <notifications@github.com> wrote:












@AngelMunoz @jbeck111 are you sure its this package and not minimap or something else?

**Comment by jbeck111** [comment]
@basarat  It's not minimap - I don't use it. But for me the problem is now gone. I was tinkering with the tsconfig.json and making the file list from the filesGlob. I saw your question and re-opened Atom to check on the problem - but it's gone. Sorry!

**Comment by MMasey** [comment]
Getting the same error, i was not experiencing this yesterday. Here's my stack trace.

Many thanks.

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.1.0
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v7.4.4


### Stack Trace

Uncaught TypeError: editor.displayBuffer.getScrollTop is not a function

```
At C:\Users\mike.masey\.atom\packages\atom-typescript\dist\main\atom\tooltipManager.js:49

TypeError: editor.displayBuffer.getScrollTop is not a function
    at showExpressionType (C:\Users\mike.masey\.atom\packages\atom-typescript\dist\main\atom\tooltipManager.js:49:45)
    at C:\Users\mike.masey\.atom\packages\atom-typescript\dist\main\atom\tooltipManager.js:40:59
```

### Commands

```
  3x -0:47.6.0 linter:lint (atom-text-editor.editor)
```

### Config

```json
{
  "core": {},
  "atom-typescript": {}
}
```

### Installed Packages

```coffee
# User
.bin, vundefined
atom-typescript, v7.4.4
emmet, v2.3.13
linter, v1.8.1
Stylus, v1.2.1

# Dev
No dev packages
```

**Comment by ericmdantas** [comment]
Same bug here.

**Comment by Chocoloper** [comment]
hey got the same bug.

atom 1.1.0
atom-typescript 7.7.2

in my TooltipManager.js the following line (49) is the problem:
"        pixelPt.top += editor.getScrollTop();"
The fix of @fcoury which made it to 7.7.2 didn't solve the problem.
Found the function in following file of the atom (stable/1.1.0) atom project.
https://github.com/atom/atom/blob/stable/src/text-editor.coffee

**Comment by Chocoloper** [comment]
Somehow i tried to look intoi the problem and started atom directly from the directory i installed it in to take a look at stdout/stderr.
Since then I had no error. Even when I started it like before again. The fix of @fcoury seems to work no. I really don't understand it but I'm happy now.

I also could reproduce it on another machine where I had the same issue.

Can somebody try it out???? I just executed ./opt/atom/atom

**Comment by enriched** [comment]
@Chocoloper Is it possible that you have multiple installations of atom on your machine?

**Comment by Chocoloper** [comment]
hmm I'm unsure about that. I installed it via following ppa: `ppa:webupd8team/atom` and also got my update via it. after the update to 1.1.0 I got the error. As I also had the old atom-typescript version with `editor.displayBuffer.getScrollTop()` I updated to 7.7.2 with the fix. The fix didn't worked for me. After the direct start via `/opt/atom/atom` instead of starting it via the atom.sh it started working. I don't think the ppa installs more than one version but I'm not sure about it.

It's strange isn't it? It was also the same procedure with my desktop and laptop (completely same stack/versions).
